app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,69 @@
+class PostsController < ApplicationController
+  before_action :set_post, only: %i[show edit update destroy]
+  before_action :authenticate_user!, only: %i[new create]
+
+  # GET /posts
+  # GET /posts.json
+  def index
+    @posts = Post.all
+  end
+
+  # GET /posts/1
+  # GET /posts/1.json
+  def show; end
+
+  # GET /posts/new
+  def new
+    @post = Post.new
+  end
+
+  # GET /posts/1/edit
+  def edit; end
+
+  # POST /posts
+  # POST /posts.json
+  def create
+    @post = Post.new(title: params[:post][:title], body: params[:post][:body], user_id: current_user.id)
+
+    respond_to do |format|
+      if @post.save
+        format.html { redirect_to posts_path, notice: 'Post was successfully created.' }
+        format.json { render :show, status: :created, location: @post }
+      else
+        format.html { render :new }
+        format.json { render json: @post.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # PATCH/PUT /posts/1
+  # PATCH/PUT /posts/1.json
+  def update
+    respond_to do |format|
+      if @post.update(post_params)
+        format.html { redirect_to @post, notice: 'Post was successfully updated.' }
```

Comment:
You are using `post_params` here but you never defined it. That's why your method is not working. Please fix it so that the user can update the post.

Reply:
You should only include: resources :posts, only: [:new, :create, :index] (screenshot from Odin)
For that  reason, I did not include  :update